src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -518,20 +522,42 @@ public CheckConstraintsArgs(CSharpCompilation currentCompilation, ConversionsBas
             }
         }
 
+        private static readonly ObjectPool<CheckConstraintsArgsBoxed> s_checkConstraintsArgsBoxedPool = new(() => new());
```

Comment:
Please don't use `new()` in the compiler layer.

Reply:
ok.